bpfd-operator/pkg/bpfd-agent/bpfd-agent.go

```diff
@@ -281,6 +268,19 @@ func (r *BpfProgramReconciler) reconcileBpfProgramConfig(ctx context.Context,
 		}
 
 		// otherwise load it
+		//
+		// TODO(astoycos) This will need to end up being a list of loadRequests
+		// if a given BpfProgramConfig selects more than one attach point
+		// (i.e if we support a pod LabelSelector for interfaces) For now
+		// we only support specifying a single BpfProgramAttachPoint in the API so
+		// there will only be a single loadRequest per BpfProgramConfig Object.
+		loadRequest, err := internal.BuildBpfdLoadRequest(BpfProgramConfig)
```

Comment:
If we're going to do it like this let's just move the `internal.BuildBpfdLoadRequest` function call into `r.loadBpfdProgram` to avoid code duplication (see my note for last case where we don't load.